client/host_queue.go

```diff
@@ -119,21 +124,33 @@ func (q *queue) Open() {
 }
 
 func (q *queue) flushEvery(interval time.Duration) {
+	// sleepForOverride used change the next sleep based on last ops rotation
+	var sleepForOverride time.Duration
 	for {
+		sleepFor := interval
+		if sleepForOverride > 0 {
+			sleepFor = sleepForOverride
+			sleepForOverride = 0
+		}
+
+		time.Sleep(sleepFor)
+
 		q.RLock()
-		state := q.state
-		q.RUnlock()
-		if state != stateOpen {
+		if q.state != stateOpen {
+			q.RUnlock()
 			return
 		}
+		lastRotateAt := q.opsLastRotatedAt
+		q.RUnlock()
 
-		time.Sleep(interval)
+		sinceLastRotate := q.nowFn().Sub(lastRotateAt)
+		if sinceLastRotate < interval {
+			// Rotated already recently, sleep until we would next consider flushing
+			sleepForOverride = interval - sinceLastRotate
+			continue
+		}
 
 		q.Lock()
-		if state != stateOpen {
```

Comment:
Yeah I'll need to add a RLock and keep the lock when writing to the `drainIn` channel.  Updating now.